packages/react/src/NavList/NavList.tsx

```diff
@@ -288,6 +290,57 @@ const Group: React.FC<NavListGroupProps> = ({title, children, sx: sxProp = defau
   )
 }
 
+export type NavListShowMoreItemProps = {
+  children: React.ReactNode
+  label?: string
+} & SxProp
+
+const ItemWithinGroup = React.createContext<{id: string} | null>(null)
+
+const ShowMoreItem = React.forwardRef<HTMLButtonElement, NavListShowMoreItemProps>(
+  ({label = 'Show more', children, ...props}, forwardedRef) => {
+    const [expanded, setExpanded] = React.useState(false)
+    const targetFocused = React.useRef(false)
+
+    const id = useId()
+    const groupId = React.useMemo(() => ({id}), [id])
+
+    React.useEffect(() => {
+      if (expanded && !targetFocused.current) {
+        const focusTarget: HTMLAnchorElement | null = document.querySelector(
+          `[data-show-more-group-id="${groupId.id}"]`,
+        )
+
+        if (focusTarget) {
+          focusTarget.focus()
+          targetFocused.current = true
+        }
+      }
```

Comment:
Wanted to ask if this could run in the `onClick` handler instead of us needing to synchronize this with an effect 👀 The benefit of tying this to the handler would be that focus would shift naturally due to interacting with the button instead of potentially being stolen if one the dependencies of the effect change that is not tied to a user interaction

Reply:
We could, but I think the only issue is the timing, as the node we want to focus once expanded doesn't exist yet 🤔. 

Something like the following works, but I'm not sure if it's counterintuitive or not 😅 

```tsx
onClick={() => {
  flushSync(() => {
    setExpanded(true)
  })
  expandItem()
}}
```